apps/admin-x-settings/src/utils/localeValidation.ts

```diff
@@ -0,0 +1,149 @@
+export const validateLocale = (value: string): string | null => {
+    const errorMessage = 'Invalid locale format. Examples: en, pt-BR, zh-Hant, sr-Cyrl, x-private';
+
+    if (!value) {
+        return 'Locale is required';
+    }
+
+    // Trim whitespace
+    const trimmedValue = value.trim();
+    
+    if (!trimmedValue) {
+        return errorMessage;
+    }
+
+    // Grandfathered irregular tags (exact matches, case insensitive)
+    const grandfatheredIrregular = [
+        'en-gb-oed', 'i-ami', 'i-bnn', 'i-default', 'i-enochian', 'i-hak',
+        'i-klingon', 'i-lux', 'i-mingo', 'i-navajo', 'i-pwn', 'i-tao',
+        'i-tay', 'i-tsu', 'sgn-be-fr', 'sgn-be-nl', 'sgn-ch-de'
+    ];
+    
+    // Grandfathered regular tags (exact matches, case insensitive)
+    const grandfatheredRegular = [
+        'art-lojban', 'cel-gaulish', 'no-bok', 'no-nyn', 'zh-guoyu',
+        'zh-hakka', 'zh-min', 'zh-min-nan', 'zh-xiang'
+    ];
```

Comment:
What is the source of these grandfathered tags? I think it would be useful to add a bit more explanation to the comments to help future code explorers